### ISSUE TITLE
Missing hint for removing the `elem` function

### ISSUE THREAD

**Issue by asr** [ndmitchell/hlint#249]
In the following example

```haskell
foo :: Char -> Char -> Bool
foo a b = a `elem` [b]
```

shouldn't HLint suggests

```
Found:
  a `elem` 'b`
Why not:
  a == b
```

?

The missing hint was suggested by @andreasabel [here](https://github.com/agda/agda/commit/2537c49d6057b770c15d6d9dba6f3dd5c7d84c44#commitcomment-17851682).

```bash
$ hlint --version
HLint v1.9.34, (C) Neil Mitchell 2006-2016
```

**Comment by ndmitchell** [comment]
Yep, that seems like a good idea!

**Comment by asr** [comment]
Note that the hint should be suggested in the general case

```haskell
foo :: Eq a => a -> a -> Bool
foo a b = a `elem` [b]
```

**Comment by ndmitchell** [comment]
HLint is not type directed, so it will definitely be the general case.

**Comment by ndmitchell** [comment]
Note that the real issue was:

```
a `elem` "b"
```

Which in HLint requires a separate hint, since its purely syntax directed.

**Comment by ndmitchell** [comment]
Implemented the list variant. I decided against the String variant for two reasons: 1) it's technically difficult to implement in HLint requiring new pieces of scaffolding; 2) the OverloadedStrings extension means it might be wrong in some circumstances.

Will be in the next release, but am not intending to make a release until something else crops up. Let me know if you'd prefer a release with it in sooner.

**Comment by asr** [comment]
> Implemented the list variant.

Thanks!

> Let me know if you'd prefer a release with it in sooner.

I can wait for the release.
